src/__tests__/hooks.test.js

```diff
@@ -654,6 +654,78 @@ describe('hooks', () => {
     expect(innerRenderCount).toEqual(1);
   });
 
+  it('does not cause unnecessary dispatches due to object literals passed to useTracking', () => {
+    let dispatchCount = 0;
+
+    const App = () => {
+      const [state, setState] = React.useState({});
+
+      useTracking(
+        {},
+        {
+          dispatchOnMount: () => {
+            dispatchCount += 1;
+          },
+        }
+      );
```

Comment:
What you have here is functionally the same but just in case this behavior ever changes do you mind updating this so that the actual `dispatch` calls are counted and not just the side effect of calling dispatchOnMount that's counted?

```suggestion
      useTracking(
        {},
        {
          dispatch,
          dispatchOnMount: () => {
            return { test: true };
          },
        }
      );
```

Then later:

```js
    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(dispatch).toHaveBeenLastCalledWith({
      test: true,
    });
```